### PR TITLE
[WV][000000] Extracted default map options

### PR DESCRIPTION
**Pull request by warrenv**
- Extracted default map options. This will require the application to
  provide them.
- Updated the demo app to define the map option on an application object
  stored on window.
- Removed unnecessary requirejs dependencies.

### CODE REVIEW

File: app/coffeescript/map.coffee [hunk 1]
```diff
@@ -27,10 +28,6 @@ define [
           draggable: true
           panControl: false
       markers:
-        mapPin:        "/assets/nonsprite/map/map_pin_red4.png"
```
Comment (c0) [inline app/coffeescript/map.coffee:28]: If you set the values to either `undefined` or `null` (can't remember which), then FlightJS makes it required.

**Comment by mmaiza** [comment]
It is ok to extract them from map. But we will need to add them to a pins mixin, we can set them as null so they are required and will raise an error if not provided.
Please resolve the conflict so I can merge.

File: app/coffeescript/map.coffee (hunk 1 above)
Comment (warrenv) [inline app/coffeescript/map.coffee:28]: thanks for the tip. I will do that in a future Pr that has pins working in the demo.

**Comment by warrenv** [comment]
I rebased off of dev and there were no merge conflicts.

**Comment by warrenv** [comment]
I think it would be easier for a developer to update a slim file that a mixin in the case of changing map assets.

**Comment by mmaiza** [comment]
The mixin allows to define the pin trait and use anywhere it is needed. The assetUrl itself can be part of the application configuration. But the paths to shadow, free, cluster are really attributes of the pin object. This in addition will allow us to define pins depending on resolutions.

**Pull request merged**